src/StepDefinition/MinkElementTrait.php

```diff
@@ -185,4 +185,59 @@ public function textAppearsTimesInElement($text, $element, $number, $selector =
     }
   }
 
+  /**
+   * @Then /^(?:|I )wait for (?P<num>\d+) "(?P<element>[^"]*)" elements to appear?$/
+   *
+   * @param string $element
+   * @param int $num
+   *
+   * @throws \Exception
+   */
+  public function iWaitForElementToAppear($num, $element) {
+    $this->spinFunction(function ($context) use ($element, $num) {
+      try {
+        $this->assertSession()->elementsCount('css', $element, intval($num));
+        return TRUE;
+      }
+      catch (ExpectationException $e) {
+        // NOOP
+      }
+      return FALSE;
+    });
+  }
+
+  /**
+   * Implementation of spin method.
+   *
+   * @param string $lambda
+   *   Function expresion.
+   *
+   * @return bool
+   *
+   * @throws \Exception
+   */
+  public function spinFunction($lambda) {
+    $tries = 0;
+    while ($tries < 30) {
+      try {
+        if ($lambda($this)) {
+          return TRUE;
+        }
+      }
+      catch (Exception $e) {
+        // Do nothing.
+      }
+      sleep(1);
+      $tries++;
+    }
+    throw new ExpectationException('Matching number of elements not found.', $this->getSession());
```

Comment:
This makes the function non-generic, thus it should be private and named non-generic.